src/dbnode/client/session.go

```diff
@@ -3423,6 +3437,7 @@ func (r *bulkBlocksResult) addBlockFromPeer(
 
 		result = r.blockOpts.DatabaseBlockPool().Get()
 		result.Reset(start, blockSize, encoder.Discard())
+		result.SetNamespaceContext(r.nsCtx)
```

Comment:
Hm, is it worth just making `nsCtx` an arg to `Reset(...)` if it's always required?

Reply:
agree.